src/Aspire.Hosting.AWS/CloudFormation/CloudFormationTemplateResource.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Publishing;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting.AWS.CloudFormation;
+
+/// <inheritdoc/>
+internal sealed class CloudFormationTemplateResource(string name, string templatePath) : CloudFormationResource(name), ICloudFormationTemplateResource
+{
+    public IDictionary<string, string> CloudFormationParameters { get; } = new Dictionary<string, string>();
+
+    /// <inheritdoc/>
+    public string TemplatePath { get; } = templatePath;
+
+    /// <inheritdoc/>
+    public ICloudFormationTemplateResource AddParameter(string parameterName, string parameterValue)
```

Comment:
strings only?

Reply:
Yes, this string eventually gets passed into the CloudFormation Parameter API that only takes strings. https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_Parameter.html